packaging/linux/common/khiops

```diff
@@ -1,56 +1,59 @@
 #!/bin/bash
 
 # Run Khiops tool
-# see the khiops-env script for full documentation on the environment variables used by Khiops
-
-
+# see the khiops_env script for full documentation on the environment variables used by Khiops
 
 error() {
     if [ -f /dev/log ]; then
         logger "$*" --tag khiops --stderr
+    else
+        echo "$*" >&2
     fi
-    if [ -n "$DISPLAY" ]
-    then
-        notify-send "Khiops" "$*" -i /usr/share/pixmaps/khiops.png -u normal;
+    if [ -n "$DISPLAY" ]; then
+        notify-send "Khiops" "$*" -i /usr/share/pixmaps/khiops.png -u normal
     fi
-    exit 1; 
+    exit 1
 }
 
-
 warning() {
     if [ -f /dev/log ]; then
         logger "$*" --tag khiops --stderr
+    else
+        echo "$*" >&2
     fi
-    if [ -n "$DISPLAY" ]
-    then
-        notify-send "Khiops" "$*" -i /usr/share/pixmaps/khiops.png -u normal;
+    if [ -n "$DISPLAY" ]; then
+        notify-send "Khiops" "$*" -i /usr/share/pixmaps/khiops.png -u normal
     fi
 }
 
-if [[ ! -f /usr/bin/khiops-env ]]; then
-    error "/usr/bin/khiops-env is missing"
+get_script_dir() {
+    SCRIPT_DIR=$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &>/dev/null && pwd)
+    echo "$SCRIPT_DIR/"
+}
+
+if [[ ! -f "$(get_script_dir)"/khiops_env ]]; then
+    echo "$(get_script_dir)/khiops_env is missing"
+    exit 1
 fi
 
-# initialize Khiops environment
-source /usr/bin/khiops-env
+# Initialize Khiops env variables
+source $(get_script_dir)/khiops_env ""
 
 # Test if batch mode from parameters
 KHIOPS_BATCH_MODE=false
-for ((i=1;i<=$#;i++)); 
-do
+for ((i = 1; i <= $#; i++)); do
     if [ ${!i} = "-b" ] || [ ${!i} = "-h" ] || [ ${!i} = "-s" ] || [ ${!i} = "-v" ]; then
-	    KHIOPS_BATCH_MODE=true
+        KHIOPS_BATCH_MODE=true
         break
     fi
-done;
-
+done
 
-if [[ ! -e $KHIOPS_PATH/MODL ]]; then
-    error "Binary not found ($KHIOPS_PATH/MODL)"
+if [[ ! -e $KHIOPS_PATH ]]; then
+    error "Binary not found ($KHIOPS_PATH)"
 fi
 
-# Test multiple possible errors with the java GUI 
-if [  "$KHIOPS_BATCH_MODE" = false ]; then
+# Test multiple possible errors with the java GUI
+if [ "$KHIOPS_BATCH_MODE" = false ]; then
```

Comment:
In environments where the Khiops GUI is not supported *and* `DISPLAY` is defined, namely in Conda, a spurious error regarding the GUI is issued (on line 61 below) in the following cases:

1. A non-existing option is passed, e.g. `khiops --foo` (but also `khiops --version` or `khiops --help` which are not supported, although `khiops -v` and `khiops -h` are)
2. An existing option is passed, but the `-b` option is not passed.

IMHO, we should:

1. Inject a `_IS_CONDA` flag from CMake via `khiops_env` which is sourced anyway by `khiops`
2. If `_IS_CONDA` is set, then:
    - for case 1. above : just display `MODL_*`  help (as the `MODL_*` binaries do)
    - for case 2. above: just display error message that GUI is not supported and ask to pass the `-b` flag (as it is done when `DISPLAY` is not set - which works for usual Docker setups)
3. Else, proceed as of now:
    - Docker:
       - if `DISPLAY` is not set, then ask to pass `-b`: OK
       - if `DISPLAY` is set, then the current behavior looks OK to me, because on those environments users can install the full `khiops` native package which provides GUI suport
4. For encapsulation purposes, `unset _IS_CONDA`.

Reply:
Latest version testing report:
- Conda / Ubuntu 22.04 : 
   - `khiops_env --env` : OK
   - `khiops -foo` : OK
   - `khiops -b -foo`: OK
   - idem `khiops_coclustering`
- Docker / Ubuntu 22.04 / nodisplay : 
   - `khiops_env --env` : OK
   - `khiops -foo` : OK, but OpenMPI-related(?) warning: `Authorization required, but no authorization protocol specified` twice at the beginning of the output (presumably related to OpenMPI running as root)
   - `khiops -b -foo`:  OK, but OpenMPI-related(?) warning: `Authorization required, but no authorization protocol specified` twice at the beginning of the output (presumably related to OpenMPI running as root)
   - `khiops_coclustering -foo`: OK
   - `khiops_coclustering -b -foo`: OK
- Docker / Rocky 9 / nodisplay : 
- Docker / Ubuntu 22.04 / display :
- Docker / Rocky 9 / display :